### PR TITLE
refactor(docs): refactor front-matter to use mapping instead of switch cases

### PR DESCRIPTION
**Pull request by RishhiB**
just a small refactor to use mapping

### CODE REVIEW

File: docs/api-markdown-documenter/front-matter.js [hunk 1]
```diff
@@ -20,95 +20,77 @@ const generatedContentNotice =
...
+function createHugoFrontMatter(apiItem, config, customRenderers, version) {
+	const associatedPackage = apiItem.getAssociatedPackage();
+	const frontMatter = {
+		title: createFrontMatterTitle(apiItem),
+		version,
+		kind: apiItem.kind,
+		members: new Map(),
+		package: associatedPackage ?
+			associatedPackage.name.replace(/"/g, "").replace(/!/g, "") :
+			"undefined",
```
Comment (Josmithr) [inline docs/api-markdown-documenter/front-matter.js:40]:
```suggestion
			undefined,
```

File: docs/api-markdown-documenter/front-matter.js [hunk 2]
```diff
@@ -20,95 +20,77 @@ const generatedContentNotice =
...
+const createFrontMatterTitle = (apiItem) => {
+	const baseTitle = apiItem.displayName.replace(/"/g, "").replace(/!/g, "");
+	const kindTitleMap = {
+		[ApiItemKind.Model]: "Package Reference",
+		[ApiItemKind.Class]: `${baseTitle} Class`,
+		[ApiItemKind.Interface]: `${baseTitle} Interface`,
+		[ApiItemKind.Package]: `${baseTitle} Package`,
+		[ApiItemKind.Namespace]: `${baseTitle} Namespace`,
+	};
+	return kindTitleMap[apiItem.kind] || baseTitle;
```
Comment (Josmithr) [inline docs/api-markdown-documenter/front-matter.js:92]: Nit
```suggestion
	return kindTitleMap[apiItem.kind] ?? baseTitle;
```

File: docs/api-markdown-documenter/front-matter.js (hunk 1 above)
Comment (Josmithr) [inline docs/api-markdown-documenter/front-matter.js:40]:
```suggestion
		package: associatedPackage?.name.replace(/"/g, "").replace(/!/g, "") 
```

File: docs/api-markdown-documenter/front-matter.js [hunk 3]
```diff
@@ -20,95 +20,77 @@ const generatedContentNotice =
...
+	const apiMembers = apiItem.kind === ApiItemKind.Package ?
+		apiItem.entryPoints[0].members :
+		apiItem.members;
 
-		const documentWriter = createDocumentWriter();
-		MarkdownRenderer.renderNode(summaryParagraph, documentWriter, {
-			customRenderers,
+	apiMembers.filter(({ displayName }) => displayName !== "")
+		.forEach(api => {
+			frontMatter.members[api.kind] = frontMatter.members[api.kind] || {};
```
Comment (Josmithr) [inline docs/api-markdown-documenter/front-matter.js:56]: Nit
```suggestion
			frontMatter.members[api.kind] = frontMatter.members[api.kind] ?? {};
```

**Review by Josmithr — APPROVED** [review]
Changes look good to me! Thanks for the cleanup!

You'll want to merge main to pick up my most recent changes here, but otherwise should be good to go! 🚀

**Comment by RishhiB** [comment]
oh no, got issues merging main in

**Pull request closed without merging**